pkg/cluster/private_registry.go

```diff
@@ -34,15 +34,31 @@ func GetPrivateRepo(cluster *v3.Cluster) *rketypes.PrivateRegistry {
 	return nil
 }
 
+// TransformProvV2RegistryCredentialsToDockerConfigJSON transforms a ProvV2 registry secret into a credentialprovider.DockerConfigJSON secret
+func TransformProvV2RegistryCredentialsToDockerConfigJSON(url string, registryCredentials *corev1.Secret) (map[string][]byte, error) {
+	username := string(registryCredentials.Data["username"])
+	password := string(registryCredentials.Data["password"])
+	authConfig := credentialprovider.DockerConfigJSON{
```

Comment:
Sorry I didn't catch this first time, but it looks like we create a `credentialprovider.DockerConfigJSON` that we immediately render to a json blob, then later we unmarshal the json, only to create an identical `credentialprovider.DockerConfigJSON` here: https://github.com/HarrisonWAffel/rancher/blob/a8c181e72bd8a7e37caa22ecc34392633115bab0/pkg/cluster/private_registry.go#L97-L105, that we then marshal to more identical json. Seems like alot of back and forth we probably could avoid`